backend/lib/endpoints/posts.js

```diff
@@ -1,53 +1,57 @@
 const httpErrors = require("http-errors");
+const Post = require("../models/Post");
```

Comment:
why did you remove this? I set it up to have a mongo connection as plugin in fastify so it connects to specific database and can be used on the server instance
This comes straight out of the Fastify documentation https://www.fastify.io/docs/latest/Getting-Started/#getting-started

Reply:
> I can't approve this if it removes the work I did on previous PR with Fastify and Mongoose integration

Really? You merged your PR ignoring eslint rules (clearly wrong and I said you to don't merge) and you can't give an approval because I changed your code?

To be honest doesn't make any sense use a plugin for mongo, because we can have less code connecting directly to the model we want. Why did you change the models folder that was working before? Complicated code should always be avoided. I have more experience in fastify and we need to be mature to do code reviews. In my PR we only have one plugin file because that will be needed for all private endpoints and it makes sense to do that. The mongo connection should always be on server file, and the code should connect to an existent mongo connection via model. Less is more, we need always to remember that.

Reply:
What do you mean by "Furthermore it looks like it's not using the correct Auth0 flow but it should use the one for Single Page Apps"? Did you read the articles I mentioned on slack? SPA flow is very insecure (Implicit Flow) and the authorization flow on the server side is the most secure flow because doesn't expose the keys on the client, and I'm very happy to debate about authentication with you

Reply:
Which ESLint rule was broken? It cannot be committed if it break an ESLint rule, and the rules can be decided by us, it's important to understand why the rules are there in the first place.
You have approved my PR and now you just removed all the work I've done to integrate Mongoose with Fastify in a much better way ? That is unacceptable.

This previous method of starting Mongoose used a global connection and needed an event emitter but I have managed to integrated it with the `createConnection()` method which nicely integrates with promises and async/await, so much better than having to rely on the event emitter separately, and it also supports multiple connections to a specific database if needed.

Furthermore I also added code to auto-load all the Mongoose schema's and instantiated them as a model based on the filename so for my that looked as a very convenient way to integrate that, you have just removed that? 

The Fastify docs clearly staate that they recommend to use plugins as much as possible so your claim "that you have more experience"  is totally irrelevant here and not a good way to discuss a code change.

Reply:
I approved your first version, then you changed the code, I said you to don't merge it and you merged. I won't discuss it furthermore with you, please @manualzuru can you help here?

Reply:
@neemiasjnr this is not true, I even had a PR with 1 commit https://github.com/manualzuru/FightPandemics.com/pull/73
And while my PR clearly states it was working on the database integration with Fastify, this change that you did is totally out of scope of the authentication and authorization.

Reply:
I'm going to jump in here just to try and make progress.

One of the main benefits of only having changes focused on a single feature/bug/etc. in a single PR is to avoid this exact situation where a controversial, unrelated change holds up other changes.

I would propose we (@neemiasjnr) removes the model related changes from this PR and opens a new PR with just those changes. We can focus #76 solely on authentication and have a rigorous debate about the other changes in their own PR. I believe other folks are waiting on the authentication changes here so getting them into `master` as soon as possible is a priority.

How does that sound to everyone?

Reply:
Sounds good to me, I also had some other remarks about the auth part but guess I;ll keep that for the other PR.